File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
for test_case in range(1):
    n = int(input())
    arr = []
    for _ in range(n):
        arr.append(list(map(int, input().split())))
    arr.sort(key=lambda x:(x[1],x[0]))
    idx = 0
    cnt = 1
    while True:
        start, end = arr[idx][0], arr[idx][1]
        flg = False
        for i in range(idx+1, len(arr)):
            if arr[i][0] >= end:
                idx = i
                flg = True
                break
        if not flg:
            break
        cnt += 1
    print(cnt)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.java
import java.util.*;
import java.lang.*;
import java.io.*;

// The main method must be in a class named "Main".
class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());

        int[][] arr = new int[n][2];

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());

            arr[i][0] = start;
            arr[i][1] = end;
        }

        Arrays.sort(arr, (a, b) -> {
            if (a[1] == b[1]) return a[0] - b[0];
            return a[1] - b[1];
        });
        
        int[] cur_arr = new int[2];
        int idx = 0;
        int ans = 0;
        
        for (int[] a : arr) {
            
            if (idx == 0) {
                cur_arr = a;
                ans = 1;
                idx++;
                continue;
            }

            if (a[0] >= cur_arr[1]) {
                cur_arr = a;
                ans++;
            }
            idx++;
        }
        System.out.println(ans);
    }
}